### PR TITLE
feat(CSS): import of relative paths

### PR DESCRIPTION
**Pull request by vchimev**
Implements https://github.com/NativeScript/NativeScript/issues/6021.

### CODE REVIEW

File: tns-core-modules/ui/styling/style-scope.ts [hunk 1]
```diff
@@ -637,6 +651,24 @@ export function resolveFileNameFromUrl(url: string, appDirectory: string, fileEx
     return null;
 }
 
+function resolveFileNameFromImport(importSource: string, fileName: string): string {
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: add comments to explain how this works and format it (missing ;, let -> const, declare every variable with separate statement, etc.)

File: tns-core-modules/ui/styling/style-scope.d.ts [hunk 2]
```diff
@@ -36,5 +36,5 @@ export class StyleScope {
     public getAnimations(ruleset: RuleSet): KeyframeAnimationInfo[];
 }
 
-export function resolveFileNameFromUrl(url: string, appDirectory: string, fileExists: (string) => boolean): string;
+export function resolveFileNameFromUrl(url: string, appDirectory: string, fileExists: (string) => boolean, importSource?: string): string;
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.d.ts]: why is this exported?

File: tns-core-modules/ui/styling/style-scope.ts [hunk 3]
```diff
@@ -109,6 +103,17 @@ class CSSSource {
         return CSSSource.fromFile(appRelativeUri, keyframes);
     }
 
+    private static uriToRelativePath(uri: string): string {
+        let appRelativeUri = uri;
+        if (appRelativeUri.startsWith("/")) {
+            var app = knownFolders.currentApp().path + "/";
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: const

File: tns-core-modules/ui/styling/style-scope.ts [hunk 4]
```diff
@@ -201,12 +206,13 @@ class CSSSource {
         const imports = this._ast["stylesheet"]["rules"].filter(r => r.type === "import");
         for (let i = 0; i < imports.length; i++) {
             const importItem = imports[i]["import"];
-
+            const importItemSource = imports[i]["position"]["source"];
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: check if `imports[i]["position"]` is not undefined before accessing `["source"]`

File: tns-core-modules/ui/styling/style-scope.ts [hunk 5]
```diff
@@ -201,12 +206,13 @@ class CSSSource {
...
             const match = importItem && (<string>importItem).match(pattern);
             const url = match && match[2];
 
             if (url !== null && url !== undefined) {
-                const cssFile = CSSSource.fromURI(url, this._keyframes);
+                const file = CSSSource.resolveCSSPathFromURL(url, importItemSource);
+                const cssFile = new CSSSource(undefined, url, file, this._keyframes, undefined);
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: use the static method `fromFile`

File: tns-core-modules/ui/styling/style-scope.ts [hunk 6]
```diff
@@ -201,12 +206,13 @@ class CSSSource {
...
         for (let i = 0; i < imports.length; i++) {
             const importItem = imports[i]["import"];
-
+            const importItemSource = imports[i]["position"]["source"];
             const match = importItem && (<string>importItem).match(pattern);
             const url = match && match[2];
 
             if (url !== null && url !== undefined) {
-                const cssFile = CSSSource.fromURI(url, this._keyframes);
+                const file = CSSSource.resolveCSSPathFromURL(url, importItemSource);
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: what will happen if file is `null`?

File: tests/app/ui/styling/style-tests.ts [hunk 7]
```diff
@@ -1409,50 +1409,72 @@ export function test_CascadingClassNamesAppliesAfterPageLoad() {
 }
 
 export function test_resolveFileNameFromUrl_local_file_tilda() {
```
Comment (sis0k0) [inline tests/app/ui/styling/style-tests.ts]: Add tests for `resolveFileNameFromUrl` with the new argument (`importSource`) as well.

File: tns-core-modules/ui/styling/style-scope.ts (hunk 3 above)
Comment (vchimev) [reply tns-core-modules/ui/styling/style-scope.ts]: Stays `let` - four lines below there is a new assignment of `appRelativeUri`. Changed `var app` to `let app`.

File: tns-core-modules/ui/styling/style-scope.ts [hunk 8]
```diff
@@ -109,6 +103,21 @@ class CSSSource {
...
     }
 
+    private static pathRelativeToApp(uri: string): string {
+        if (!uri.startsWith("/")) {
+            return uri;
+        }
+
+        const appPath = knownFolders.currentApp().path;
+        if (!uri.startsWith(appPath)) {
+            traceWrite("${uri} does not start with ${appPath}", traceCategories.Error, traceMessageType.error);
```
Comment (sis0k0) [inline tns-core-modules/ui/styling/style-scope.ts]: Backticks (`) instead of quotes (")

**Pull request merged**